src/odin/ingestion/afc/afc_archive.py

```diff
@@ -238,9 +237,22 @@ def download_csv(self, download_jobs: ApiCounts) -> None:
             "sidTo": str(sid_to),
         }
         r = self.make_request(url, fields=fields, preload_content=False)
-        with gzip.open(BytesIO(r.data)) as gdata:
-            with open(csv_path, mode="wb") as writer:
-                writer.write(gdata.read())
+
+        csv_path = os.path.join(self.tmpdir, f"{sid_from}.csv")
+        gz_path = f"{csv_path}.gz"
+
+        # download csv.gz file (as stream)
+        with open(gz_path, mode="wb") as gz_write:
+            shutil.copyfileobj(r, gz_write)
+        r.release_conn()
```

Comment:
Actually read the urllib3 docs: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#streaming-and-io